### ISSUE TITLE
Documentation, and attitude.

### ISSUE THREAD

**Issue by larryhynes** [jekyll/jekyll-assets#207]
The 'old' jekyll assets (which was great - thank you @ixti) used to have very useful documentation in its README - see [here](https://github.com/jekyll/jekyll-assets/tree/legacy#jekyllassets). I find the current README to be a pale imitation - do you have any plans to improve it?

Now, lest I be accused of leading @envygeeks on a [wild goose chase](https://github.com/jekyll/jekyll-assets/issues/155) I shall try to be clear and specific:

* Your opening line is:

> Jekyll 3 assets is an asset pipeline using Sprockets 3 to build especially for Jekyll 3. It utilizes new features of both Sprockets and Jekyll to achieve a clean and extensible assets platform for Jekyll.

Might I suggest you allude to those among us who had been using jekyll-assets for some time, and how we might go about availing of this new version? Like how it might / will / could affect our existing setups? You might earn some bonus points by making it clear if we have to *do* anything to avail of all this sprocketry - like, install it or something.

* You continue:

> When you are using a Gemfile and bundler you need to do nothing special to get Jekyll Assets to work, it will automatically load itself and work with Jekyll when you bundle install and run Jekyll through bundle exec. However, when you have globally installed Gems (gem install jekyll-assets) then in your _config.yml do:

Would it be churlish of me to point out that it could be seen as good practice to use `bundle install` and `bundle exec`? There may be some people visiting your page that are less familiar than you with the internal workings of Jekyll. (One would hate them to get the impression that their contribution - and use of Jekyll - was in some way unwelcome or of a lesser degree.)

* Digest options

There are some poor souls, labouring in their ignorance, who have used jekyll-assets in the past to, well, digest their assets. (I'm not a proud man; I will admit I am among their number.) A question was asked about this in an issue [here](https://github.com/jekyll/jekyll-assets/issues/194) to which you reply:

> Cache busting is unnecessary because Sprockets 3.0 tracks all assets that you use and busts it themselves as they see fit using a much more robust algorithm that we would use (or did use.) We have no plans to cache outside of sprockets as it increases complexity for very little benefit.

Again - and do please excuse me if I'm 'bangin on' here - is there anything that us inarticulate troglodytes must to to avail of all this whizz-bang sprocketness? Could I humbly suggest that you jot a quick note for us and leave it somewhere?

I'll not go on.

In short: a lot of people have a jekyll-assets that currently works a certain way. It appears, though I am in no way certain, that we will now have a jekyll-assets that will work a different way. Please document same.

Ordinarily I would be moved to draft some suggestions and send a pr, but your tone suggests to me that you might just like to do it yourself and not have to deal with what I imagine you would exasperatedly view as my impertinence.

Clear enough?

**Comment by envygeeks** [comment]
Let me get this right, you want me to be nice to you while approaching me the way you did?

**Comment by larryhynes** [comment]
No, I want you to document the change from jekyll-assets to 'new' jekyll-assets. I have absolutely no expectation of you being nice.

**Comment by envygeeks** [comment]
First let me get this out of the way, I am a generally very receptive and accepting person unless argument are not properly posed and explanations are broad and then yes I do get terse and I will not change that.  I am a very opinionated person and I will not change that, but my mind can be changed in the face of a good argument and t hat can be demonstrated all over the place. I do not get paid to do this, I do this because I want to help you.  I do not need to help you, I didn't need to update this and add Sprockets 3, I could have just kept it a private fork for my clients and let them get the benefits.

I'm not required to go on any goose chase, I am not required to do research for a feature you want.  This is open source, if you want a feature and you want *me* to implement it, you do the research, you explain ti clearly and you make sure I understand it or I will just refuse the idea because again, you do not pay me to do the research, you do not pay me to maintain this, I maintain this for you for *free*.

Let me address your quote of "banging on", here: http://www.usingenglish.com/reference/phrasal-verbs/bang+on+about.html

> The 'old' jekyll assets (which was great - thank you @ixti) used to have very useful documentation in its README - see here. I find the current README to be a pale imitation - do you have any plans to improve it?

Coming in and implying my software is bad because *you* think my documentation is bad is not a very good way to convince me to consider your opinion as valid.  I don't know what's wrong with it, and the hundreds of clients I've had use that documentation seem to not have a problem with it. So how about you elaborate.

> Might I suggest you allude to those among us who had been using jekyll-assets for some time, and how we might go about availing of this new version? Like how it might / will / could affect our existing setups? You might earn some bonus points by making it clear if we have to do anything to avail of all this sprocketry - like, install it or something.

This is utter trash.  Time and time again I have tried to make features backwards compatible if I felt it was important and the user implied it was important.  I've even gone out of my way to implement features that I said wouldn't be implemented and even gone out of my damn way to implement features ahead of their schedule for users because I felt they felt it was important to them.

> Would it be churlish of me to point out that it could be seen as good practice to use bundle install and bundle exec? There may be some people visiting your page that are less familiar than you with the internal workings of Jekyll. (One would hate them to get the impression that their contribution - and use of Jekyll - was in some way unwelcome or of a lesser degree.)

What? How exactly does what I said imply their contribution is less important?  You are sending mixed messages here, and this is exactly why I end up being terse.

> Again - and do please excuse me if I'm 'bangin on' here - is there anything that us inarticulate troglodytes must to to avail of all this whizz-bang sprocketness? Could I humbly suggest that you jot a quick note for us and leave it somewhere?

Yes, and I'm happy to accept any changes you feel are incorrect to the documentation  most of the time I will not turn down any contribution unless the code is highly unoptimized (and I'll work on fixing that with you) or completely unacceptable (at which time I will generally ask you to come back with better code -- but that has only ever happened once.)  I am normally the first person to vote :+1: to a contribution in Jekyll and Jekyll-Assets even if I don't want it or like it (and I generally keep the "don't like it" comments to myself.) There are times though where I think things can be handled better so I turn away a contribution and there is nothing wrong with that, I'm not required to accept every pull request and if I turn away a contribution I explain why I think it's not right.

I am a programmer who typically gets hired by other programmers so sometimes I write documentation tersely with the assumption everybody will get it, that doesn't mean I think it's right.  I am opinionated about writing code, not writing documentation.

As a matter of fact there were a few times in tickets where somebody closed a ticket after I explained things and I explictly held the ticket open to correct the documentation.  So what exactly are you implying?

---

There was nothing wrong with what I said at all either.  We don't need to offer you the ability to bust assets the way old Jekyll Assets did, in development we don't digest assets so they are always fresh if Sprockets 3 wants them to be (but you can force it -- as said in the documentation) and in production Sprockets keeps track of whether your asset changes and busts it's own cache creating a new digest for you and we always digest the URL in production.  Even through our Proxies and soon Transformers.

There was a second a few days ago and it's on my note in the backlog to teach people how to use JEKYLL_ENV to do all that automatically but I haven't had time to get to it because this isn't my job I do this for you for free.

> Ordinarily I would be moved to draft some suggestions and send a pr, but your tone suggests to me that you might just like to do it yourself and not have to deal with what I imagine you would exasperatedly view as my impertinence.

That's not true at all, and you are assuming based on one instance (and that had nothing to do with this and if it did, that's exactly why I asked them to be clearer because I had no idea it was about that even after multiple conversational attempts) and what are the first 4 letters of assumption?  I always work with people on features and documentation and I am very receptive of pull requests so if you feel something is not explained clearly please submit a pull request and I'l review it and more often than not be very quick about getting it in for you so that other people don't have the same confusion.

**Comment by larryhynes** [comment]
There was a Jekyll plugin called jekyll-assets that lived at [https://github.com/ixti/jekyll-assets](https://github.com/ixti/jekyll-assets). I and many others used it. That page now redirects to this jekyll-assets. Nowhere have you mentioned that. That is poor documentation and you should fix it.

For those of use who did use the old jekyll-assets, you do not document how use of the new jekyll-assets might affect us. That is poor documentation and you should fix it.

There is nothing terse about accusing someone who has not explained themselves to your satisfaction of 'banging on'. Nor is there anything terse, or well-documented, about redirecting a url for an existing plugin to a page that does not even mention that plugin nor its developer nor the work that went into it nor its users. That is many things, but it is not terse.

As to your coming down from on high to gift your vaunted expertise to me for free? I liked the old plugin. Perhaps I will like the new one. (It's a bit of a moot point, as I'm in the process of changing to Hakyll.) Either way, you do not do this for me. You do it for your own complex reasons. I am as free to use or not use it as you are to provide or not provide it. I will not be a hostage to your martyrdom. If it brings you joy and pleasure to provide this – and deal with those who might wish to use it – please carry on. If not, I suggest you stop.

**Comment by envygeeks** [comment]
I'm closing this ticket nad locking it, you are making wild ass assumptions and attacking me for no reason, I don't need to take your shit.  Have a nice life.
